metrics/exporter.go

```diff
@@ -80,6 +173,10 @@ func getCurMetricsConfig() *metricsConfig {
 func setCurMetricsConfig(c *metricsConfig) {
 	metricsMux.Lock()
 	defer metricsMux.Unlock()
+	setCurMetricsConfigUnlocked(c)
+}
+
+func setCurMetricsConfigUnlocked(c *metricsConfig) {
```

Comment:
How about changing this one to `setCurMetricsConfig` and the above one to `setCurMetricsConfigForTesting` and move it to a test file since it is only used in test now?

Reply:
It might be best if we just leave the existing functions as locked functions, and introduce these new unlocked versions for the exporter's use.

We have some other non-testing components in the metrics package that are using the existing getCurMetricsConfig(). Those calls do require the lock.

Only tests use the setter functions, but it would be very strange to have getCurMetricsConfig() use a lock, but then have setCurMetricsConfig() not.

Reply:
To me it's more strange to add testing only functions in non-test files. @vagababov  WDYT?

Reply:
Since there are existing non-test components that expect thread-safe getCurMetricsConfig, I don't think we can make setCurMetricsConfig unlocked. It would be very confusing for anyone trying to use that function or read the code in the future to reason about thread safety.

I think it's common to implement Getters/Setters in pairs for API completeness, even if they're not both used immediately.

But I can move setCurMetricsConfig to a separate test_utils file for now to make it clear that it's only used in tests?

Reply:
I'm going to leave all the default get/set functions as locked to prevent future confusion.